odoo/addons/test_runtime/fields.py

```diff
@@ -0,0 +1,41 @@
+from odoo import fields, models
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class SelectionStr(str):
+    def init_vals(self, selection, field_name):
+        self.__selection = selection
+        self.__field_name = field_name
+
+    def __eq__(self, other):
+        if isinstance(other, str) and other not in self.__selection:
+            _logger.warning(
+                "Wrong check for selection field on %s: %s\n"
+                "Valid values are: %s" % (
+                    self.__field_name,
+                    other,
+                    ', '.join(self.__selection),
+                )
+            )
+        return super().__eq__(other)
```

Comment:
Optimize: if `self == other`, you don't need to check anything.  And selection values may be put in a set once.
```suggestion
    def __eq__(self, other):
        if super().__eq__(other):
            return True
        try:
            selection = self.__selection
        except AttributeError:
            selection = self.__selection = set(self.__field.get_values(self.__env))
        if isinstance(other, str) and other not in selection:
            _logger.warning("Comparing %s to invalid value %r", self.__field, other)
        return False
```